src/Linters/HHClientLinter.hack

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+use type Facebook\HHAST\{BaseLinter};
+use namespace Facebook\TypeAssert;
+use namespace HH\Lib\{C, Vec};
+
+final class HHClientLinter extends BaseLinter {
+
+  const type TConfig = shape();
```

Comment:
The active lints need to be configurable; perhaps:

```
shape(
  ?'ignore_except' => keyset<int>,
  ?'ignore' => keyset<int>,
 )
```

Both would default to empty; if something is present in both, that should be a runtime error.

Additionally, there should probably be a hardcoded list of lints to already ignore, e.g. 5583 - await in a loop - is known buggy, and has a pure Hack implementation in HHAST and Facebook www already. The current list is HackLintRule::HACK_LINT_IGNORED_CODES in FB WWW, mostly being known-bad linters that should be fixed at some point, but for 5583/AWAIT_IN_LOOP, this is permanently replaced by the linters written in Hack so blocking it here (and in FB) is a band aid - the right fix is probably to delete that one from the ocaml code.

Reply:
I will implement the config in another PR as this PR is already too large